File: app/src/main/java/com/fjsdkqwj/pfdioewjnsd/ui/adapter/MineItemAdapter.java
package com.fjsdkqwj.pfdioewjnsd.ui.adapter;

import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.fjsdkqwj.pfdioewjnsd.R;
import com.fjsdkqwj.pfdioewjnsd.model.MineItemModel;
import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;

import java.util.List;

public class MineItemAdapter extends BaseQuickAdapter<MineItemModel, BaseViewHolder> {

    private ItemClickListener itemClickListener;

    public MineItemAdapter(int layoutResId, @Nullable List<MineItemModel> data) {
        super(layoutResId, data);
    }

    @Override
    protected void convert(@NonNull BaseViewHolder helper, MineItemModel item) {
        helper.setText(R.id.item_title_tv, item.getItemTitle());
        ImageView itemImg = helper.getView(R.id.item_img);
        itemImg.setImageResource(item.getImgRes());
        helper.getView(R.id.parent_ll).setOnClickListener(v -> {
            if (itemClickListener != null){
                itemClickListener.onItemClicked(helper.getLayoutPosition());
            }
        });
    }

    public void setItemClickListener(ItemClickListener itemClickListener){
        this.itemClickListener = itemClickListener;
    }

    public interface ItemClickListener {
        void onItemClicked(int position);
    }
}

File: app/src/main/java/com/qwsdbsdvdsnd/dfmcvndfhdfsdfwe/ui/adapter/MineItemAdapter.kt
package com.qwsdbsdvdsnd.dfmcvndfhdfsdfwe.ui.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.qwsdbsdvdsnd.dfmcvndfhdfsdfwe.R
import com.qwsdbsdvdsnd.dfmcvndfhdfsdfwe.mode.MineItemModel

class MineItemAdapter(context: Context, mList: ArrayList<MineItemModel>) :
    RecyclerView.Adapter<MineItemAdapter.MineItemViewHolder>() {

    private val context: Context? = context

    private val mList: MutableList<MineItemModel>? = mList

    private var itemClickListener: MineItemAdapter.ItemClickListener? = null

    inner class MineItemViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val item_title_tv: TextView = itemView.findViewById(R.id.item_title_tv)
        val item_img: ImageView = itemView.findViewById(R.id.item_img);
        val parent_ll: View = itemView.findViewById(R.id.parent_ll);
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MineItemViewHolder {
        val view: View =
            LayoutInflater.from(context).inflate(R.layout.adapter_mine_list_litm, parent, false)
        return MineItemViewHolder(view)
    }

    override fun onBindViewHolder(holder: MineItemViewHolder, position: Int) {
        val model: MineItemModel? = mList?.get(position)
        model?.let {
            holder.item_title_tv.text = it.itemTitle
            holder.item_img.setImageResource(it.imgRes)
        }
        holder.parent_ll.setOnClickListener {
            if (itemClickListener != null) {
                itemClickListener!!.onItemClicked(position)
            }
        }
    }

    override fun getItemCount(): Int = mList!!.size

    fun setItemClickListener(itemClickListener: ItemClickListener) {
        this.itemClickListener = itemClickListener
    }

    interface ItemClickListener {
        fun onItemClicked(position: Int)
    }

}